openfermioncirq/gates/fermionic_simulation.py

```diff
@@ -87,6 +88,90 @@ def state_swap_eigen_component(x: str, y: str, sign: int = 1, angle: float = 0):
     return component
 
 
+def fermionic_simulation_gates_from_interaction_operator(
+        operator: openfermion.InteractionOperator):
+    r"""
+    Given $H = \sum_a H_a$, returns gates
+    $\left\{G_a\right\} = \left\{e^{i H_a\right\}$.
+
+    Args:
+        operator: The interaction operator ($H$).
+
+    Returns: A dict from tuples of mode indices to gates.
+    """
+    n_qubits = operator.n_qubits
+
+    gates: Dict[Tuple[int, ...], cirq.Gate] = {}
+
+    if operator.constant:
+        gates[()] = operator.constant
+    for p in range(n_qubits):
+        coeff = operator.one_body_tensor[p, p]
+        if coeff:
+            gates[(p,)] = cirq.Z**(coeff / np.pi)
+    for modes in itertools.combinations(range(n_qubits), 2):
+        gate: Optional[InteractionOperatorFermionicGate] = (
+            QuadraticFermionicSimulationGate.from_interaction_operator(
+                operator=operator, modes=modes))
+        if gate:
+            gates[modes] = gate
+    for modes in itertools.combinations(range(n_qubits), 3):
+        gate = CubicFermionicSimulationGate.from_interaction_operator(
+            operator=operator, modes=modes)
+        if gate:
+            gates[modes] = gate
+    for modes in itertools.combinations(range(n_qubits), 4):
+        gate = QuarticFermionicSimulationGate.from_interaction_operator(
+            operator=operator, modes=modes)
+        if gate:
+            gates[modes] = gate
+    return gates
+
+
+def interaction_operator_from_fermionic_simulation_gates(
```

Comment:
What's the use case of this function? Is there a reason you don't expose it, even though it is not hidden? I think it needs a more descriptive name, maybe something like `sum_of_interaction_operator_gate_generators`.

Reply:
The only use case I have at this point is for testing.